### PR TITLE
Fix skin parser not stripping whitespace before parsing

### LINKED ISSUE #12051: Skin using wrong colors, instead of just blue and white
**Describe the bug:** 
There's a skin I use on osu stable that normally uses a blue and white color scheme. When I imported my skins to osu lazer, the skin wouldn't use the blue and white color scheme, and would only use really vibrant red green yellow and blue colors for sliders/ hit circles. Ive tried reimporting the skins over and over to no avail. Here is the skin link: https://www.mediafire.com/folder/bxwutdn2cpr5y/-Gawr_Gura-  heres some screen shots of the skin (how its supposed to look: 
image
image

I have modified the cursor in the pack, but i redownloaded the original unedited version and deleted the skin from osu stable and osu lazer, and it still had the same issue. I'm back to using the modified version.
**Screenshots or videos showing encountered issue:** 
image
image
The screenshots look a bit weird( pixely / no aa) but its perfect ingame, besides the colors. I also am using the original skin in lazer, not the edited one i use in stable, as i haven't taken the time to import my skins again from stable, as it didn't work last time. But when i tried with the edited skin, it didn't work, like i said.
Heres how it looks in lazer: 
**osu!lazer version:** 
2021.312.0
**Logs:**
[performance.log](https://github.com/ppy/osu/files/6160573/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6160574/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6160575/updater.log)
[database.log](https://github.com/ppy/osu/files/6160576/database.log)

### PR DESCRIPTION
**Pull request by peppy**
Closes #12051.

### CODE REVIEW

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs [hunk 1]
```diff
@@ -36,6 +36,8 @@ protected override void ParseStreamInto(LineBufferedReader stream, T output)
                 if (ShouldSkipLine(line))
                     continue;
 
+                line = line.Trim();
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyDecoder.cs:39]: I'm fairly certain this should be `line = StripComments(line).TrimEnd()` if anything. Especially don't trim the start - that breaks storyboard decoding.

Also, can remove calls to `StripComments()` from other areas.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyDecoder.cs:39]: Right, didn't realise this was also being used to parse storyboard content as well.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 2]
```diff
@@ -67,36 +67,34 @@ protected override void ParseStreamInto(LineBufferedReader stream, Beatmap beatm
...
+                    handleGeneral(line);
                     return;
 
                 case Section.Editor:
-                    handleEditor(strippedLine);
+                    handleEditor(line);
                     return;
 
                 case Section.Metadata:
                     handleMetadata(line);
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:81]: Perhaps the reason why it was doing it locally was because of this?

**Review by smoogipoo — APPROVED** [review]
lgtm

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:81]: Yeah.. on checking the usages of the `Editor` section on stable there doesn't look to be valid reasoning behind this either way.

**Pull request merged**